llvm/lib/FileCheck/FileCheck.cpp

```diff
@@ -2486,6 +2489,14 @@ static bool ValidatePrefixes(StringRef Kind, StringSet<> &UniquePrefixes,
              << "check and comment prefixes: '" << Prefix << "'\n";
       return false;
     }
+    for (StringRef Directive : Directives) {
```

Comment:
What's the motivation for this part of the change? Has it caused issues?

If we are doing this, a test should be added to the FileCheck test suite.

Reply:
I think a prefix that looks like a prefix + directive is very misleading and can easily lead to bug as the test gets updated. E.g. a prefix CHECK-NEXT could easily go wrong. That sounds like a good change to me but should have a lit test indeed.

Reply:
> Has it caused issues?

@klensy reported that
https://github.com/llvm/llvm-project/blame/main/lld/test/ELF/arm-exidx-shared.s used `// RUN: llvm-objdump -s --triple=armv7a-none-linux-gnueabi %t2 | FileCheck --check-prefix=CHECK-EXTAB-NEXT %s`.

`CHECK-EXTAB-NEXT` was very confusing. I fixed it.